master/buildbot/www/hooks/github.py

```diff
@@ -73,7 +74,13 @@ def _get_payload(self, request):
             if hash_type != 'sha1':
                 raise ValueError('Unknown hash type: %s' % (hash_type,))
 
-            mac = hmac.new(self._secret, msg=content, digestmod=sha1)
+            _secret = self._secret
+            if isinstance(_secret, text_type):
```

Comment:
There is already ascii2unicode here:
https://github.com/buildbot/buildbot/blob/master/master/buildbot/util/__init__.py#L178

for coherency, I would call it

    def unicode2utf8(value):
        if isinstance(value, text_types):
            value = value.encode('utf-8')
        return value

and put it just after ascii2unicode